api.php

```diff
@@ -9,6 +9,7 @@
   '/^(?:\\/\\/)?\s*(?:instance|class)?\s*\(?(' . PRE_IDENT . ')\)?\s*(?:infix[lr]?\s+\d\s*(?:\\/\\/)?)?(?:\s+a\s+)?::.*$/mi');
 
 function search_doc(&$r, $name, $libraries){
+	$filtermod = isset($_GET['mod']);
```

Comment:
To separate model and controller, it would be preferable to add a fourth argument to `search_doc` which is either a list of modules that can be accepted, or `NULL` if the user doesn't wish to search on this.